routers/org/setting.go

```diff
@@ -85,12 +85,30 @@ func SettingsPost(ctx *context.Context, form auth.UpdateOrgSettingForm) {
 	org.Description = form.Description
 	org.Website = form.Website
 	org.Location = form.Location
-	org.Visibility = form.Visibility
 	org.RepoAdminChangeTeamAccess = form.RepoAdminChangeTeamAccess
+
+	visibilityChanged := form.Visibility != org.Visibility
+	org.Visibility = form.Visibility
+
 	if err := models.UpdateUser(org); err != nil {
 		ctx.ServerError("UpdateUser", err)
 		return
 	}
+
+	// update forks visibility
+	if visibilityChanged {
+		if err := org.GetRepositories(models.ListOptions{Page: 1, PageSize: org.NumRepos}); err != nil {
```

Comment:
This seems quite inefficient as there is no guarantee that NumRepos is at all small, so this should probably be paged. (Similarly for the get ForkedRepos function - but you don't have to fix that in this PR.)

This should also be moved into the UpdateUser(org) function.

I've also just realised the implications of UpdateRepository - we're literally going to iterate across all forks of forks of repositories etc. I think that might need some more thought.

Reply:
But `UpdateUser` is just a silly wrapper for `e.ID(u.ID).AllCols().Update(u)`

Reply:
>I've also just realised the implications of UpdateRepository - we're literally going to iterate across all forks of forks of repositories etc. I think that might need some more thought.

Well yes, we need to update visibility for every fork and their subsequent forks.

Reply:
I don't know if I'm too tired but... shouldn't this query bring the forked repositories (which will belong to users) rather than org's?

Reply:
Also, what happens if userB forks from userA who forked from orgA?

Reply:
@guillep2k It brings all repositories of organization and then executes `UpdateRepository` on them, which in turn will fetch all forks of it and execute `UpdateRepository` on these forks which will fetch all forks of it and then execute `UpdateRepository` on these forks... until all layers of forks are done and then we are onto next repo and it all starts again, until we are done with org repos and then we return.